moveit_planners/ompl/ompl_interface/src/model_based_planning_context.cpp

```diff
@@ -959,6 +968,62 @@ void ompl_interface::ModelBasedPlanningContext::unregisterTerminationCondition()
   ptc_ = nullptr;
 }
 
+void ompl_interface::ModelBasedPlanningContext::omplPlannerStatusToMoveItErrorCode(
+    og::SimpleSetupPtr ompl_simple_setup, moveit_msgs::msg::MoveItErrorCodes& res)
+{
+  ompl::base::PlannerStatus ompl_status = ompl_simple_setup->getLastPlannerStatus();
+  if (ompl_status == ompl::base::PlannerStatus::UNKNOWN)
+  {
+    RCLCPP_WARN(LOGGER, "Motion planning failed for an unknown reason");
+    res.val = moveit_msgs::msg::MoveItErrorCodes::PLANNING_FAILED;
+  }
+  else if (ompl_status == ompl::base::PlannerStatus::INVALID_START)
+  {
+    RCLCPP_WARN(LOGGER, "Invalid start state");
+    res.val = moveit_msgs::msg::MoveItErrorCodes::PLANNING_FAILED;
```

Comment:
Do you think we could add something like INVALID_START to the MoveItError codes?
Right now we have `START_STATE_IN_COLLISION` and `START_STATE_VIOLATES_PATH_CONSTRAINTS` but none both do not correspond to OMPL's message definition, yet I think it is helpful to have that information in the error code as well.

You could add a TODO here and open an issue in MoveIt2 so we can add this later

Reply:
I'm glad you suggested this because I had the same thought. I started a PR here:

https://github.com/ros-planning/moveit_msgs/pull/146